src/pages/templates/index.tsx

```diff
@@ -0,0 +1,64 @@
+import { PageId } from "page-id";
+import { ThirdwebNextPage } from "utils/types";
+import TemplateWrapper, {
+  filterTemplates,
+} from "../../components/templates/Wrapper";
+import { GetServerSidePropsContext } from "next";
+import { TEMPLATE_TAGS, TemplateTagId } from "data/templates/tags";
+import { TEMPLATE_DATA } from "data/templates/templates";
+
+const title = "Web3 Templates for Websites & Apps";
+const description =
+  "Start building with a library of quick-start templates for web3 apps and websites — for NFTs, marketplaces, and more. Get started.";
+
+const Templates: ThirdwebNextPage = (props: TemplatePageProps) => {
+  const { _defaultTagIds, _defaultTemplates, _defaultKeyword } = props;
+  return (
+    <>
+      <TemplateWrapper
+        title={title}
+        description={description}
+        data={_defaultTemplates}
+        showFilterMenu={true}
+        _defaultTagIds={_defaultTagIds}
+        _defaultKeyword={_defaultKeyword}
+      >
+        <></>
+      </TemplateWrapper>
+    </>
+  );
+};
+
+Templates.pageId = PageId.Templates;
+
+export default Templates;
+
+type TemplatePageProps = {
+  _defaultTagIds: TemplateTagId[];
+  _defaultTemplates: (typeof TEMPLATE_DATA)[number][];
+  _defaultKeyword?: string;
+};
+
+export async function getServerSideProps(context: GetServerSidePropsContext) {
```

Comment:
can't use `getServerSideProps` for this, this would mean every single visitor to the page would trigger server rendering for this, instead we should pre-generate this page like we do for the tagId as well, for filtering since we only have a very limited count of templates we can just actually filter cleint-side => we do not need server rendering for this page at all